packages/sol-cov/src/coverage_manager.ts

```diff
@@ -0,0 +1,166 @@
+            _.some(sourceRanges, range => utils.isRangeInside(range.location, location)),
+        );
+        branchCoverage[branchId] = isCovered;
+    }
+    const statementCoverage: StatementCoverage = {};
+    for (const statementId of _.keys(coverageEntriesDescription.statementMap)) {
+        const statementDescription = coverageEntriesDescription.statementMap[statementId];
+        const isCovered = _.some(sourceRanges, range => utils.isRangeInside(range.location, statementDescription));
+        statementCoverage[statementId] = isCovered;
+    }
+    const functionCoverage: FunctionCoverage = {};
+    for (const fnId of _.keys(coverageEntriesDescription.fnMap)) {
+        const functionDescription = coverageEntriesDescription.fnMap[fnId];
+        const isCovered = _.some(sourceRanges, range => utils.isRangeInside(range.location, functionDescription.loc));
+        functionCoverage[fnId] = isCovered;
+    }
+    const partialCoverage = {
+        [contractData.sources[fileIndex]]: {
+            ...coverageEntriesDescription,
+            l: {}, // It's able to derive it from statement coverage
+            path: fileName,
+            f: functionCoverage,
+            s: statementCoverage,
+            b: branchCoverage,
+        },
+    };
+    return partialCoverage;
+}
+
+export class CoverageManager {
+    private _traceInfoByAddress: { [address: string]: TraceInfo[] } = {};
+    private _contractsData: ContractData[] = [];
+    private _txDataByHash: { [txHash: string]: string } = {};
+    private _getContractCodeAsync: (address: string) => Promise<string>;
+    constructor(
+        artifactsPath: string,
+        sourcesPath: string,
+        networkId: number,
+        getContractCodeAsync: (address: string) => Promise<string>,
+    ) {
+        this._getContractCodeAsync = getContractCodeAsync;
+        this._contractsData = collectContractsData(artifactsPath, sourcesPath, networkId);
+    }
+    public setTxDataByHash(txHash: string, data: string): void {
+        this._txDataByHash[txHash] = data;
+    }
+    public appendTraceInfo(address: string, traceInfo: TraceInfo): void {
+        if (_.isUndefined(this._traceInfoByAddress[address])) {
+            this._traceInfoByAddress[address] = [];
+        }
+        this._traceInfoByAddress[address].push(traceInfo);
+    }
+    public async writeCoverageAsync(): Promise<void> {
+        const finalCoverage = await this._computeCoverageAsync();
+        const jsonReplacer: null = null;
+        const numberOfJsonSpaces = 4;
+        const stringifiedCoverage = JSON.stringify(finalCoverage, jsonReplacer, numberOfJsonSpaces);
+        fs.writeFileSync('coverage/coverage.json', stringifiedCoverage);
+    }
+    private async _computeCoverageAsync(): Promise<Coverage> {
+        const collector = new Collector();
+        for (const address of _.keys(this._traceInfoByAddress)) {
+            if (address !== constants.NEW_CONTRACT) {
```

Comment:
Assign this to a descriptive variable name please. Then we might not need the comment below?